### DISCUSSION TITLE
Configure Emacs to view cmd output and screen scrollback

### DISCUSSION THREAD

**Discussion by xuhdev** [kovidgoyal/kitty#8148] [Show and tell]
In `~/.local/bin/emacs-pager.sh`:

```bash
#!/bin/sh

t=$(mktemp "/tmp/$USER/.emacs-pager.XXXXXXX") || exit 1
cat - > $t
echo 'reading into emacs...'
emacsclient --tty --alternate-editor '' "$t"
rm -f -- $t
```

In `kitty.conf`:

```
map kitty_mod+g launch --stdin-source=@last_cmd_output --type=overlay ${HOME}/.local/bin/emacs-pager.sh
map kitty_mod+j launch --stdin-source=@first_cmd_output_on_screen --type=overlay ${HOME}/.local/bin/emacs-pager.sh
map kitty_mod+h launch --stdin-source=@screen_scrollback --type=overlay ${HOME}/.local/bin/emacs-pager.sh
```

Now `kitty_mod+g`, `kitty_mod+j`, `kitty_mod+h` start emacs and show last command output/first command output on screen/the whole scrollback!

Credit: Script inspired by https://write.as/mnmlmnl/tid-setting-emacsclient-up-as-a-pager-with-kitty

**Comment by kovidgoyal** [comment]
If you use bash or zsh instead you can have the shell manage the temp file for you with something like 

```sh
emacsclient --tty --alternate-editor '' <(cat /dev/stdin)
```

The you dont need emacs-pager.sh at all you can just do:

```
map kitty_mod+g launch --stdin-source=@last_cmd_output --type=overlay zsh -c "emacsclient --tty --alternate-editor '' <(cat /dev/stdin)"
```

**Reply by xuhdev** [reply]
I'm getting the "File exists, but cannot be read" error on Linux. Not sure why though...
